pyproject.toml

```
@@ -0,0 +1,199 @@
+name = 'sdgym'
+version = '0.7.1.dev0'
+license = { text = 'BSL-1.1' }
+requires-python = '>=3.8,<3.12'
+readme = 'README.md'
+urls = { 'Homepage' = 'https://github.com/sdv-dev/SDGym' }
+
+dependencies = [
+    'appdirs>=1.3,<2',
+    'boto3>=1.15.0,<2',
+    'botocore>=1.18,<2',
+    'compress-pickle>=1.2.0,<3',
+    'humanfriendly>=8.2,<11',
+    "numpy>=1.20.0,<2;python_version<'3.10'",
+    "numpy>=1.23.3,<2;python_version>='3.10'",
+    "pandas>=1.1.3;python_version<'3.10'",
+    "pandas>=1.3.4;python_version>='3.10' and python_version<'3.11'",
+    "pandas>=1.5.0;python_version>='3.11'",
+    'psutil>=5.7,<6',
+    "scikit-learn>=0.24,<2;python_version<'3.10'",
+    "scikit-learn>=1.1.3,<2;python_version>='3.10'",
+    "scipy>=1.5.4,<2;python_version<'3.10'",
+    "scipy>=1.9.2,<2;python_version>='3.10'",
+    'tabulate>=0.8.3,<0.9',
+    "torch>=1.8.0;python_version<'3.10'",
+    "torch>=1.11.0;python_version>='3.10' and python_version<'3.11'",
+    "torch>=2.0.0;python_version>='3.11'",
+    'tqdm>=4.15,<5',
+    'XlsxWriter>=1.2.8,<4',
+    'rdt>=1.6.1,<2.0',
+    'sdmetrics>=0.11.0,<1.0',
+    'sdv>=1.3.0,<2',
+]
+
+[project.entry-points]
+sdgym = { main = 'sdgym.cli.__main__:main' }
+
+[project.optional-dependencies]
+
+dask = ['dask', 'distributed']
+
+test = [
+    'pytest>=3.4.2',
+    'pytest-cov>=2.6.0',
+    'jupyter>=1.0.0,<2',
+    'rundoc>=0.4.3,<0.5',
+]
+
+dev = [
+    # dask
+    'dask',
+    'distributed',
+
+    # test
+    'pytest>=3.4.2',
+    'pytest-cov>=2.6.0',
+    'jupyter>=1.0.0,<2',
+    'rundoc>=0.4.3,<0.5',
+
+    # general
+    'bumpversion>=0.5.3,<0.6',
+    'pip>=9.0.1',
+    'watchdog>=0.8.3,<0.11',
+
+    # style check
+    'flake8>=3.7.7,<4',
+    'flake8-absolute-import>=1.0,<2',
+    'flake8-builtins>=1.5.3,<1.6',
+    'flake8-comprehensions>=3.6.1,<3.7',
+    'flake8-debugger>=4.0.0,<4.1',
+    'flake8-docstrings>=1.5.0,<2',
+    'flake8-eradicate>=1.1.0,<1.2',
+    'flake8-fixme>=1.1.1,<1.2',
+    'flake8-mock>=0.3,<0.4',
+    'flake8-multiline-containers>=0.0.18,<0.1',
+    'flake8-mutable>=1.2.0,<1.3',
+    'flake8-expression-complexity>=0.0.9,<0.1',
+    'flake8-print>=4.0.0,<4.1',
+    'flake8-pytest-style>=1.5.0,<2',
+    'flake8-quotes>=3.3.0,<4',
+    'flake8-sfs>=0.0.3,<0.1',
+    'flake8-variables-names>=0.0.4,<0.1',
+    'dlint>=0.11.0,<0.12',
+    'isort>=4.3.4,<5',
+    'pandas-vet>=0.2.3,<0.3',
+    'pep8-naming>=0.12.1,<0.13',
+    'pydocstyle>=6.1.1,<6.2',
+
+    # fix style issues
+    'autoflake>=1.1,<2',
+    'autopep8>=1.4.3,<2',
+
+    # distribute on PyPI
+    'twine>=1.10.0,<4',
+    'wheel>=0.30.0',
+
+    # Advanced testing
+    'coverage>=4.5.1,<6',
+    'tox>=2.9.1,<4',
+    'importlib-metadata>=3.6',
+
+    # Invoke
+    'invoke',
+]
+
+all = [
+    # dask
+    'dask',
+    'distributed',
+
+    # test
+    'pytest>=3.4.2',
+    'pytest-cov>=2.6.0',
+    'jupyter>=1.0.0,<2',
+    'rundoc>=0.4.3,<0.5',
```

You can reference previous dependencies as such:
```suggestion
        "sdgym[dask,test]",
```

---

Yes, done in [7fac59d](https://github.com/sdv-dev/SDGym/pull/269/commits/7fac59d7ad872f32dc3b7307944cfe1980724f17)